Billing.API/Services/Invoice/InvoiceService.cs

```diff
@@ -16,31 +16,42 @@ public class InvoiceService : IInvoiceService
     {
         private readonly ILogger<InvoiceService> _logger;
         private readonly IOptions<InvoiceProviderOptions> _options;
+        private readonly ISapServiceSettingsService _sapServiceSettingsService;
 
-        public InvoiceService(ILogger<InvoiceService> logger, IOptions<InvoiceProviderOptions> options)
+        public InvoiceService(ILogger<InvoiceService> logger, IOptions<InvoiceProviderOptions> options, ISapServiceSettingsService sapServiceSettingsService)
         {
             _logger  = logger;
             _options = options;
+            _sapServiceSettingsService = sapServiceSettingsService;
         }
 
         public async Task<PaginatedResult<InvoiceListItem>> GetInvoices(string clientPrefix, int clientId, int page, int pageSize, string sortColumn, bool sortAsc)
         {
-            var dt = await GetInvoiceRecords(clientPrefix, clientId);
-
-            var invoices = dt.Select().Select(dr => new InvoiceListItem(
-                dr.Field<string>("DocumentType"),
-                dr.Field<string>("DocumentNumber"),
-                clientPrefix,
-                clientId.ToString(),
-                dr.Field<DateTime>("CreateDate").ToDateTimeOffSet(),
-                dr.Field<DateTime>("DocDueDate").ToDateTimeOffSet(),
-                dr.Field<DateTime>("SendDate").ToDateTimeOffSet(),
-                dr.Field<string>("DocCur"),
-                dr.Field<decimal>("DocTotal").ToDouble(),
-                dr.Field<decimal>("PaidToDate").ToDouble(),
-                $"invoice_{dr.Field<DateTime>("SendDate"):yyyy-MM-dd}_{dr.Field<int>("AbsEntry")}.{dr.Field<string>("FileExt")}",
-                dr.Field<int>("AbsEntry")))
-                .AsQueryable();
+            var sapSystems = _options.Value.ConfigsBySystem;
+            IQueryable<InvoiceListItem> invoices = new List<InvoiceListItem>().AsQueryable();
+
+            //Get invoices for all supported sap system because the client can be in more than one sap system.
+            //For example: One client exists in SAP AR and then it changes the billing system to QBL then the client is created in the SAP US
+            //for it reason we need get the invoices for all supported sap system.
```

Comment:
In general, we use spaces after `//`

```suggestion
            // Get invoices for all supported sap system because the client can be in more than one sap system.
            // For example: One client exists in SAP AR and then it changes the billing system to QBL then the client is created in the SAP US
            // for it reason we need get the invoices for all supported sap system.
```

Reply:
good, I will add the whitespaces.

Reply:
Done